monitor.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"time"
+)
+
+// Monitor is an interface implemented by a value that can record metrics about
+// the daemon.
+/*
+type Monitor interface {
+	// RecordServingTime records the amount of time it took to completely
+	// handle a request to the system (acceptance to full response).
+	RecordServingTime(time.Duration)
+	// IncrementAssignment increases the count of number of times an
+	// assignment was given a specific location.
+	IncrementLocation(string)
+	// IncrementErrorAssignment increases the count of the number of errors
+	// that occurred while processing an incoming assignment.
+	IncrementErrorHandlerAssignment()
+	// IncrementErrorRollout increases the count of the number of errors
+	// that occurred while processing a rollout value.
+	IncrementRollout(error)
+}
+*/
+type Monitor interface {
+	RecordServe(error)
+	RecordServingTime(time.Duration)
+	RecordHandling(string, error)
+	RecordRolloutUpdate(error)
+}
+
+// A DogStatsDMonitor represents a proxy for sending metrics to Datadog using
+// the namespace prefix "maxwellsdaemon.".
+type DogStatsDMonitor struct {
```

Comment:
All the implementations of the common interfaces (`Handler`, `Server`, etc) have been public.  Not that it matters either way since this is all `package main`.